business_requirement_deliverable_cost/models/business.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class BusinessRequirementResource(models.Model):
+    _inherit = "business.requirement.resource"
+
+    sale_price_unit = fields.Float(
+        string='Sales Price',
+        readonly=True,
+        groups='business_requirement_deliverable_cost.\
+group_business_requirement_estimation',
+    )
+    sale_price_total = fields.Float(
+        compute='_compute_sale_price_total',
+        string='Total Revenue',
+        groups='business_requirement_deliverable_cost.\
+group_business_requirement_estimation',
+    )
+    unit_price = fields.Float(
+        groups='business_requirement_deliverable_cost.\
+group_business_requirement_cost_control',
+    )
+    price_total = fields.Float(
+        groups='business_requirement_deliverable_cost.\
+group_business_requirement_cost_control',
+    )
+
+    @api.multi
+    @api.depends('sale_price_unit', 'qty')
+    def _compute_sale_price_total(self):
+        for resource in self:
+            resource.sale_price_total = resource.sale_price_unit * resource.qty
+
+    @api.multi
+    @api.onchange('product_id')
+    def product_id_change(self):
+        for resource in self:
+            super(BusinessRequirementResource, resource).product_id_change()
+            deliverable_project = \
+                resource.business_requirement_deliverable_id.project_id
+            if deliverable_project.pricelist_id and \
+                    deliverable_project.partner_id and resource.uom_id:
+                product = resource.product_id.with_context(
+                    lang=deliverable_project.partner_id.lang,
+                    partner=deliverable_project.partner_id.id,
+                    quantity=resource.qty,
+                    pricelist=deliverable_project.pricelist_id.id,
+                    uom=resource.uom_id.id,
+                )
+                resource.sale_price_unit = product.price
+
+    @api.onchange('uom_id', 'qty')
+    def product_uom_change(self):
+        for resource in self:
+            if not resource.uom_id:
+                resource.sale_price_unit = 0.0
+                return
+            deliverable_project = \
+                resource.business_requirement_deliverable_id.project_id
+            if deliverable_project.pricelist_id and \
+                    deliverable_project.partner_id:
+                product = resource.product_id.with_context(
+                    lang=deliverable_project.partner_id.lang,
+                    partner=deliverable_project.partner_id.id,
+                    quantity=resource.qty,
+                    pricelist=deliverable_project.pricelist_id.id,
+                    uom=resource.uom_id.id,
+                )
+                resource.sale_price_unit = product.price
```

Comment:
@elicoidal  I think it will take into account, let's leave this check when we do functional testing. LGTM